src/pdn/src/techlayer.cpp

```diff
@@ -168,6 +170,25 @@ int TechLayer::snapToManufacturingGrid(odb::dbTech* tech, int pos, bool round_up
   return pos;
 }
 
+bool TechLayer::checkIfManufacturingGrid(int value, utl::Logger* logger, const std::string& type) const
+{
+  auto* tech = layer_->getTech();
+  if (!tech->hasManufacturingGrid()) {
+    return true;
+  }
+
+  const int grid = tech->getManufacturingGrid();
+
+  if (value % grid != 0) {
+    if (logger != nullptr) {
```

Comment:
When would we not have a logger?

Reply:
I think I originally had a check that didn't have it, but you are correct, it is always present, so I could remove that check.

Reply:
ok please do so